homeassistant/components/huawei_lte/quality_scale.yaml

```diff
@@ -0,0 +1,76 @@
+rules:
+  # Bronze
+  action-setup: done
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: todo
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
```

Comment:
Are there devices without serial number?

Reply:
Possibly, or we may not be able to get one for all devices due to firmware differences, or whatnot. We explicitly fail async_setup_entry if we cannot get one though.

Reply:
Is that still the case after all this time

Reply:
Unfortunately old buggy devices aren't going anywhere, and a lot of them practically never receive any firmware updates. So I'm afraid checks like these are there to stay in one form or another -- if for nothing else, in order for us to be able to say that we can't work with the device.

Then again insisting to always have a serial number is arguably unnecessary, as setups with more than one Huawei LTE device are rare, and in setups having just one we don't really _need_ the serial number as there's nothing to distinguish the only device from. That's essentially what we used to have back in time, when the URL to the device was the identifier, but that had other issues and I'd rather not go back there.

Reply:
That's fine and I agree, but I wanted to get the context as I never did much with huawei so to get the best assessment it's usually just a good back and forth :)